core/templates/dev/head/domain/collection/read-only-collection-backend-api.service.ts

```diff
@@ -17,129 +17,143 @@
  * about collections from the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+import { AppConstants } from 'app.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
 
 // TODO(bhenning): For preview mode, this service should be replaced by a
 // separate CollectionDataService implementation which returns a local copy of
 // the collection instead. This file should not be included on the page in that
 // scenario.
-angular.module('oppia').factory('ReadOnlyCollectionBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'COLLECTION_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, UrlInterpolationService,
-      COLLECTION_DATA_URL_TEMPLATE) {
-    // Maps previously loaded collections to their IDs.
-    var _collectionCache = [];
-    var _collectionDetailsCache = [];
-
-    var _fetchCollection = function(
-        collectionId, successCallback, errorCallback) {
-      var collectionDataUrl = UrlInterpolationService.interpolateUrl(
-        COLLECTION_DATA_URL_TEMPLATE, {
-          collection_id: collectionId
-        });
-
-      $http.get(collectionDataUrl).then(function(response) {
-        var collection = angular.copy(response.data.collection);
-        _cacheCollectionDetails(response.data);
+@Injectable({
+  providedIn: 'root'
+})
+export class ReadOnlyCollectionBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService) {}
+  private _collectionCache = [];
+  private _collectionDetailsCache = [];
+
+  private _fetchCollection(
+      collectionId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    var collectionDataUrl = this.urlInterpolationService.interpolateUrl(
+      AppConstants.COLLECTION_DATA_URL_TEMPLATE, {
+        collection_id: collectionId
+      });
+
+    this.http.get(collectionDataUrl).toPromise()
+      .then((response: any) => {
+        var collection = cloneDeep(response.collection);
+        this._cacheCollectionDetails(response);
         if (successCallback) {
           successCallback(collection);
         }
-      }, function(errorResponse) {
+      }, (errorResponse) => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(errorResponse);
```

Comment:
The `.data` part is not required here?

Reply:
No, we don't need. I've added `.error` now.